jujugui/static/gui/src/app/components/profile/invoice-list/invoice-list.js

```diff
@@ -0,0 +1,155 @@
+/* Copyright (C) 2017 Canonical Ltd. */
+'use strict';
+
+const PropTypes = require('prop-types');
+const React = require('react');
+
+const BasicTable = require('../../basic-table/basic-table');
+const Spinner = require('../../spinner/spinner');
+
+/**
+  eact component used to display a list of the users invoices in their profile.
+*/
+class ProfileInvoiceList extends React.Component {
+  constructor() {
+    super();
+    this.xhrs = [];
+    this.state = {
+      data: [{
+        number: 100001,
+        status: 'Paid',
+        chargedTo: 'card ending 1234',
+        date: '12/05/2017'
+      }, {
+        number: 100002,
+        status: 'Declined: will retry',
+        chargedTo: 'card ending 1231',
+        date: '12/05/2017'
+      }, {
+        number: 100003,
+        status: 'Paid',
+        chargedTo: 'card ending 1232',
+        date: '14/05/2017'
+      }, {
+        number: 100004,
+        status: 'Paid',
+        chargedTo: 'card ending 1233',
+        date: '13/05/2017'
+      }, {
+        number: 100005,
+        status: 'Paid',
+        chargedTo: 'card ending 1234',
+        date: '12/05/2017'
+      }],
+      loading: false
+    };
+  }
+
+  componentWillMount() {
+    const user = this.props.user;
+    if (user) {
+      this._fetchInvoices(user);
+    }
+  }
+
+  componentWillUnmount() {
+    this.xhrs.forEach(xhr => {
+      xhr && xhr.abort && xhr.abort();
+    });
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const props = this.props;
+    if (props.user !== nextProps.user) {
+      this._fetchInvoices(nextProps.user);
+    }
+  }
+
+  /**
+    Fetch the users invoices
+    @param {String} user The external user name in the format "user@external".
+  */
+  _fetchInvoices(user) {
+    // @todo Fetch invoices for this user
+  }
+
+  /**
+    Prevents the default actions on the link and navigates to the specific invoice
+    for the supplied id via changeState.
+    @param {String} path The GUI invoice path to navigate to.
+    @param {Object} e The click event.
+  */
+  _navigateToInvoice(path, e) {
+    e.preventDefault();
+    e.stopPropagation();
+    // this.props.changeState({profile: null, invoice: path, hash: null});
+  }
+
+  render() {
+    let content;
+    if (this.state.loading) {
+      content = (<Spinner />);
+    } else {
+      const path = 'invoice/id';
```

Comment:
I think that we'll want the invoices to be in the profile hash not as a root path on the baseURL

Reply:
Sure, this is just a placeholder for now - actually linking each item entry to a real invoice is my next task.